src/Stratis.Bitcoin.Features.LightWallet/LightWalletSyncManager.cs

```diff
@@ -245,33 +254,55 @@ public void SyncFromDate(DateTime date)
         {
             this.logger.LogTrace("({0}:'{1}')", nameof(date), date);
 
+            if (this.syncWalletDate == DateTime.MinValue || this.walletTip.Header.BlockTime.LocalDateTime > date)
+            {
+                this.logger.LogTrace("Setting new wallet sync from date to {0}; wallet's previous sync from date was {1}.", date, this.syncWalletDate);
+                this.syncWalletDate = date;
+            }
+            else
+            {
+                this.logger.LogTrace("Ignoring request to sync from date {0} as the wallet is already syncing from older date {1}. Subsequent call to SyncFromHeight might amend the block hight.", date, this.walletTip.Header.BlockTime.LocalDateTime);
+            }
+
+            // Check if the task is already running and don't start another asyncloop call if we are already running one.
+            if (this.syncFromDateAsyncLoop != null)
+                return;
```

Comment:
Exit log

Reply:
Done

Reply:
` this.logger.LogTrace("(-)[SYNCFROMDATEASYNCLOOP_NOT_NULL]");`